### ISSUE TITLE
Closing a file should activate the most recently used document

### ISSUE THREAD

**Issue by ygoe** [notepad-plus-plus/notepad-plus-plus#885]
When closing a document by middle-clicking its tab header, the document to the left side in the tab bar is activated.

Instead, the most recently used tab should be activated, just like when pressing Ctrl+Tab.

I have configured MRU tab order and it works in other cases. Version 6.8.3.

**Comment by h-h-h-h** [comment]
:+1:

**Comment by dg9ngf** [comment]
Also, closing a tab through middle-click should focus the other document editor area, not keep the focus on the tab itself as it is now.

**Comment by sasumner** [comment]
See:
https://notepad-plus-plus.org/community/topic/14146/why-doesn-t-a-tab-close-return-you-to-tab-most-recently-worked-in
and:
https://notepad-plus-plus.org/community/topic/14678/why-doesn-t-a-tab-close-return-you-to-tab-most-recently-worked-in-part-2

**Comment by MetaChuh** [comment]
we are closing this issue due to its age and multiple newer notepad++ versions since it was filed.

please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html).
if your issue persists, and it is still of relevance to you, please feel free to file a new issue.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
```
[<kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd>](https://notepad-plus-plus.org/community/)

**Comment by ygoe** [comment]
@MetaChuh Not fixed, still highly annoying every day! Please reopen and assign a priority so that it isn't forgotten about again for the next 3.5 years.

**Comment by sasumner** [comment]
>still highly annoying every day

Every ***day*** ??!  How about "every few minutes, so many dozens of times each day"??

@SinghRajenM Since your PRs seem to get accepted, and you are taking your victory lap for the "file status auto detection" fix (hooray!)...how about taking this one on?

**Comment by ygoe** [comment]
There is a related bug, can't find its number, that the cursor is gone when a document was closed by middle-clicking its tab header. No keyboard input will be possible anymore. Only after switching documents and back, keyboard input is working again. Can somebody find that issue and link it here?

**Comment by sasumner** [comment]
Reopening with intent to create a PR based upon "gstavi" 's earlier patch for this, see here:  https://github.com/gstavi/npp/blob/master/patch/0005-Use-MRU-when-choosing-new-file-to-activate-after-clo.patch

Here's a scenario:  

Start with this setup:

![image](https://user-images.githubusercontent.com/30118311/64076918-eb857c80-cc98-11e9-99ba-9a0cadfcd500.png)

Visit the tabs in this order: `new3`, `new5`, `new7`, `new9`, `new2`, `new4`, `new6`, `new8`.  Active tab is now `new8`.

Execute `Close`.  Active tab should now be `new6`.
Execute `Close`.  Active tab should now be `new4`.
Execute `Close`.  Active tab should now be `new2`.
Execute `Close`.  Active tab should now be `new9`.
Execute `Close`.  Active tab should now be `new6`.
Execute `Close`.  Active tab should now be `new7`.
Execute `Close`.  Active tab should now be `new5`.
Execute `Close`.  Active tab should now be `new3`.
Execute `Close`.  Active tab should now be `new1`.

Actual behavior:

Execute `Close`.  Active tab is now `new`.
Execute `Close`.  Active tab is now `new`.
Execute `Close`.  Active tab is now `new`.
Execute `Close`.  Active tab is now `new`.

**Comment by sasumner** [comment]
Execute `Close`.  Active tab is now `new`.

**Comment by sasumner** [comment]
So this proposed new feature would not work jump between the primary and secondary views to get to a most-recently-used tab.  With several open tabs in each view, closing a tab would activate the most-recently-used tab in the same view, which may not be the tab the user most-recently visited (if that happened to be in the other view).  Is that behavior good, or no?

**Comment by Ekopalypse** [comment]
From my point of view, yes, if I close a tab in e.g. view1 I do **not** expect to get the focus on view0 afterwards, but get the focus on the most-recently used one in view1.

**Comment by donho** [comment]
@sasumner 
> Is that behavior good, or no?

For me it's good. Are you available to take in charge for creating a PR?

**Comment by sasumner** [comment]
@donho, to be honest, this is probably the #1 thing that annoys me about Notepad++.  So, yes, I would be thrilled to do a PR to change it.

**Comment by Shassk** [comment]
Nice job. But how do I turn it off now?
I didn't ask for this. And this behavior is unacceptable to me.
Example: I redact 10 source code files which implement some structural part of my projecf from time referencing to my TODO list file (which is **always** opened), to my main file (which is also always opened) and to my file with all the custom error codes of the project.
And when I'm done with this part - I need to close those 10 files which I opened last. One by one. Instead of switching all the time to those 3 files I have to keep opened.
But npp team in their infinite wisdom decided to not separate settings for switch tab and close tab orders for some reason.
So now I can turn it off only by turning off MRU for Ctrl+Tab whi I use all the time (otherwise navigating in 40+ tabs becomes a nightmare).
Should I open a new ticket? Or are there any ways to achieve this behavior already?

**Comment by ygoe** [comment]
Hm, strange, I never wanted the previous behaviour from before this feature was added. For me, it's a lot more useful to get back to where I was just before than to whatever happens to be next to the closed document. That's how different people and their workflows are. I guess we need an option for that.

BTW, the selected documents are not that random. You actually have a great influence on their selection just by switching to them before. So, in a way, it's your fault if the wrong document is selected after closing. ;-) Just open them one after the other and then close them again, nothing will jump around.

**Comment by Shassk** [comment]
> I guess we need an option for that.

I didn't ask for more than that.

> BTW, the selected documents are not that random. You actually have a great influence on their selection just by switching to them before. So, in a way, it's your fault if the wrong document is selected after closing. ;-) Just open them one after the other and then close them again, nothing will jump around.

For example I need to change how shop orders are edited in our CRM. For that I need opened 3+ files in products microservice (controller, model class with methods and 1+ files of interfaces) and 3+ amount for orders services, maybe even twice of that (order itself + changing part of an order). And now multiply products by 3 since we have 3 types with different product entities (normal, bundles, constructed).
Plus sometimes I open other files to get already working piece of code from another entity and change it to fit what I do at the moment.
Can you imagine how many times do I need to opend and close all that stuff if I'd do it after one-two actions?
So no, MRU becomes a mess either way with this amount of files.
I with I work on just a couple of files like I did back in the days with C++ and TensorRT.

**Comment by ss005** [comment]
@ygoe 
I understand your point, but I think it really depends on use case and the number of files we are dealing with.

I use npp for mostly pasting some temporary texts or opening some temporary files and hence I end up having many unsaved files open, maybe 50 or more at a time :P
So sometimes I need to find some text I pasted yesterday and I just randomly click on tabs or use find in files, open some files, found the text I was looking for, cool.

Now at some point I realize that there are too many tabs open and maybe recently I created too many unimportant tabs, I should close and discard some of them (new50, new49...new40 etc) which I don't need anymore.
I start from right side and look if the file content I don't need, close it, 
but here the problems comes I closed the rightmost file and then focus suddenly jumps to some random file instead of the adjacent rightmost.
Maybe my use case is weird, but it is what it is. :)

It'll be really nice to have an option.
Thanks!
